crates/swc_ecma_transforms_proposal/src/lib.rs

```diff
@@ -1,11 +1,27 @@
 #![deny(clippy::all)]
 #![allow(clippy::vec_box)]
 
+use serde::{Deserialize, Serialize};
+
 pub use self::{
     decorators::decorators, export_default_from::export_default_from,
     import_assertions::import_assertions,
 };
 
+#[derive(Debug, Default, Clone, Copy, Serialize, Deserialize)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub enum DecoratorVersion {
```

Comment:
Babel 8 keeps only the newest and the oldest versions; I think we should also keep just two versions.

Keeping the latest version is sensible, but another that leverages TypeScript's legacy decorators (experimentalDecorators) is preferable.

It is necessary to specify the exact version. SWC references the TypeScript version with the date `2021-12`, whereas Babel uses `2018-09`.

The TypeScript documentation states that it adopts a Stage 2 version, but it doesn't specify the exact year and month of the draft.

Reply:
But some users will complain that removal of implementations is a breaking change, and I'm not sure if it's fine to remove an option.

Reply:
Please only remove an option in a major version 😂